plugins/terminal.koplugin/main.lua

```diff
@@ -1,57 +1,402 @@
+            {
+                {
+                    text = _("Copy"),
+                    enabled = item.editable,
+                    callback = function()
+                        UIManager:close(shortcut_shortcuts_dialog)
+                        if self._manager.shortcuts_dialog ~= nil then
+                            UIManager:close(self._manager.shortcuts_dialog)
+                            self._manager.shortcuts_dialog = nil
+                        end
+                        self._manager:copyCommands(item)
+                    end
+                },
+                {
+                    text = _("Delete"),
+                    enabled = item.deletable,
+                    callback = function()
+                        UIManager:close(shortcut_shortcuts_dialog)
+                        if self._manager.shortcuts_dialog ~= nil then
+                            UIManager:close(self._manager.shortcuts_dialog)
+                            self._manager.shortcuts_dialog = nil
+                        end
+                        self._manager:deleteShortcut(item)
+                    end
+                }
+            }}
+        }
+        UIManager:show(shortcut_shortcuts_dialog)
+        return true
+    end
+end
+
+function Terminal:copyCommands(item)
+    local new_item = {
+        text = item.text .. " (copy)",
+        commands = item.commands
+    }
+    table.insert(self.shortcuts, new_item)
+    UIManager:show(InfoMessage:new{
+        text = _("Shortcut copied"),
+        timeout = 2
+    })
+    self:saveShortcuts()
+    self:manageShortcuts()
+end
+
+function Terminal:editCommands(item)
+    local edit_dialog
+    edit_dialog = InputDialog:new{
+        title = T(_('Edit commands for "%1"'), item.text),
+        input = item.commands,
+        width = Screen:getWidth() * 0.9,
+        para_direction_rtl = false, -- force LTR
+        input_type = "string",
+        allow_newline = true,
+        cursor_at_end = true,
+        fullscreen = true,
+        buttons = {{{
+                  text = _("Cancel"),
+                  callback = function()
+                      UIManager:close(edit_dialog)
+                      edit_dialog = nil
+                      self:manageShortcuts()
+                  end,
+              }, {
+                  text = _("Save"),
+                  callback = function()
+                      local input = edit_dialog:getInputText()
+                      UIManager:close(edit_dialog)
+                      edit_dialog = nil
+                      if input:match("[A-Za-z]") then
+                          self.shortcuts[item.nr]["commands"] = input
+                          self:saveShortcuts()
+                          self:manageShortcuts()
+                      end
+                  end,
+              }}},
+    }
+    UIManager:show(edit_dialog)
+    edit_dialog:onShowKeyboard()
+end
+
+function Terminal:editName(item)
+    local edit_dialog
+    edit_dialog = InputDialog:new{
```

Comment:
```suggestion
    local edit_dialog = InputDialog:new{
```

Reply:
I've done this on purpose. In the past I used the construction you suggest, but then UIManager:close(edit_dialog) in the callback functions didn't work. Or am I wrong?

Reply:
You're not wrong :) you actually have to do it that way when you reference it inside some included callbacks - which is the case here. (But try to not do it when it's not needed, it's kinda a clue about it happening or not.)